syntax = "proto3";
 
package starter.v1;
 
option go_package = ".;v1";
 
import "google/api/annotations.proto";
import "protoc-gen-openapiv2/options/annotations.proto";

option (grpc.gateway.protoc_gen_openapiv2.options.openapiv2_swagger) = {
    info: {
        title: "Ava";
        version: "1.0";
        contact: {
            name: "langame";
            url: "https://langa.me";
            email: "contact@langa.me";
        };
    };

    schemes: HTTP;
    schemes: HTTPS;
    consumes: "application/json";
    produces: "application/json";
};

message ConversationStarterRequest {
    string input = 1;
}
 
message ConversationStarterResponse {
    string output = 1; 
}
 
service ConversationStarterService {
    rpc GetConversationStarter(ConversationStarterRequest) returns (ConversationStarterResponse) {
        option (google.api.http) = {
            post: "/api/conversation/starter/v1"
            body: "*"
        };
    }
}